### PR TITLE
feat: modification as per review #2921

### PR DESCRIPTION
**Pull request by prajapati-kaushik**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If this relates to a card, please include a link to the card here.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/linkorb/.github/blob/master/CONTRIBUTING.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added/updated necessary documentation in the README.md or doc/ directories (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .env.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+UPR_URL:
+  description: Upr DSN url
+  required: true
+  type: url
+  exmaple: https://username:password@upr.io
+
+UPR_CACHE:
+  description: Upr  cache path. optional - should use in-memory cache if unspecified
```
Comment (joostfaassen) [inline .env.yaml]: 1. Duplicated space
2. The description does not match the definition of this variable

File: composer.json [hunk 2]
```diff
@@ -27,7 +27,8 @@
     "require": {
         "symfony/cache": "^5.0",
         "guzzlehttp/guzzle": "^6.5",
-        "symfony/dotenv": "^5.0"
+        "symfony/dotenv": "^5.0|^4.0",
+        "linkorb/envoi": "^1.0"
```
Comment (joostfaassen) [inline composer.json]: https://criterion.linkorb.com/linkorb/php-standards/items/165

File: example/metadata.php [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+// .env validate it.
+$envFilename = __DIR__.'/../.env';
+$envMetaFile = __DIR__.'/../.env.yaml';
+Envoi::init($envFilename, $envMetaFile);
+
+$hashCode = $argv[1];
+
+if (count($argv)!=2) {
+    echo "Please pass 1 parameter: hashCode\n";
+    exit();
```
Comment (joostfaassen) [inline example/metadata.php]: When exiting on error, it should be `exit(-1)`.

File: src/Client.php [hunk 4]
```diff
@@ -22,7 +24,13 @@ public function __construct(string $url, string $username, string $password, Fil
 
     public static function createFromEnv(): self
     {
-        $cache = new FilesystemAdapter(getenv('UPR_CACHE_PATH'));
+        if ($uprCache = getenv('UPR_CACHE')) {
+            if (!file_exists($uprCache)) {
+                throw new \RuntimeException('Directory not exists: '.$uprCache);
+            }
+        }
```
Comment (joostfaassen) [inline src/Client.php:32]: If unspecified, use an in memory/array cache instead of file cache

File: src/Client.php [hunk 5]
```diff
@@ -2,8 +2,10 @@
 
 namespace Upr\Client;
 
-use GuzzleHttp\Client  as GuzzleClient;
+use GuzzleHttp\Client as GuzzleClient;
 use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+//use Psr\SimpleCache\CacheInterface;
+use Symfony\Contracts\Cache\CacheInterface;
```
Comment (joostfaassen) [inline src/Client.php:8]: Please use the psr interface instead. It is universal, and more implementations exist besides symfony.

**Review by joostfaassen — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (prajapati-kaushik) [reply composer.json]: @joostfaassen 
I move packages into  "require-dev" as per my understanding.

Comment (joostfaassen) [reply composer.json]: @kaushikindianic these packages are now in `require`. `require-dev` is empty

**Review by joostfaassen — CHANGES_REQUESTED** [review]

**Pull request merged**